src/platform_impl/linux/wayland/window/shim.rs

```diff
@@ -53,6 +54,9 @@ pub enum WindowRequest {
     /// Request decorations change.
     Decorate(bool),
 
+    /// Request decorations change.
+    CsdConfig(FrameConfig),
```

Comment:
I'd probably change it to `CsdThemeVariant(crate::window::Theme)`. (Note you should export `Theme`, full path is for clarity here).